packages/pn-commons/src/utils/notification.utility.ts

```diff
@@ -436,7 +500,7 @@ function populateMacroSteps(parsedNotification: NotificationDetail) {
     for (const timelineElement of status.relatedTimelineElements) {
       populateMacroStep(parsedNotification, timelineElement, status, acceptedStatusItems);
     }
-    // order step by time
+    // order step by time, latest first
```

Comment:
Maybe we can do the shift inside this loop and before reorder?!

Reply:
Done, even when I don't like the new code. In my opinion, the `populateMacroSteps` became quite unreadable. I inserted all the comments that came out of my mind.